crates/ruff_db/src/system/memory_fs.rs

```diff
@@ -654,6 +654,30 @@ enum WalkerState {
     Nested { path: SystemPathBuf, depth: usize },
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+fn now() -> FileTime {
+    FileTime::now()
+}
+
+#[cfg(target_arch = "wasm32")]
+fn now() -> FileTime {
+    // Copied from FileTime::from_system_time()
+    let time = web_time::SystemTime::now();
+
+    time.duration_since(web_time::UNIX_EPOCH)
+        .map(|d| FileTime::from_unix_time(d.as_secs() as i64, d.subsec_nanos()))
+        .unwrap_or_else(|e| {
+            let until_epoch = e.duration();
+            let (sec_offset, nanos) = if until_epoch.subsec_nanos() == 0 {
+                (0, 0)
+            } else {
+                (-1, 1_000_000_000 - until_epoch.subsec_nanos())
+            };
+
+            FileTime::from_unix_time(-1 * until_epoch.as_secs() as i64 + sec_offset, nanos)
+        })
+}
```

Comment:
This is a bit annoying but re-implementing a memory file system just because of time also felt overkill.